airflow/kubernetes/pod_launcher.py

```diff
@@ -178,12 +178,17 @@ def monitor_pod(self, pod, get_logs):
         if self.extract_xcom:
             result = self._extract_xcom(pod)
             self.log.info(result)
-            result = json.loads(result)
-        self.istio.handle_istio_proxy(self.read_pod(pod))
+            try:
+                _result = json.loads(result)
+            except:
+                self.log.error(f"Failed to JSON decode xcom: {result}")
         while self.pod_is_running(pod):
+            self.istio.handle_istio_proxy(self.read_pod(pod))
             self.log.info('Pod %s has state %s', pod.metadata.name, State.RUNNING)
             time.sleep(SleepConfig.POD_RUNNING_POLL)
-        return self._task_status(self.read_pod(pod)), result
+        if result is None:
+            raise AirflowException('Failed to extract xcom from pod: {}'.format(pod.metadata.name))
```

Comment:
this is here to raise the error that was raised previously inside _extract_xcom, but only after handling istio